newconnector_example_test.go

```diff
@@ -0,0 +1,129 @@
+package mssql_test
+
+import (
+	"context"
+	"database/sql"
+	"flag"
+	"fmt"
+	"log"
+
+	mssql "github.com/denisenkom/go-mssqldb"
+)
+
+var (
+	debug         = flag.Bool("debug", false, "enable debugging")
+	password      = flag.String("password", "", "the database password")
+	port     *int = flag.Int("port", 1433, "the database port")
+	server        = flag.String("server", "", "the database server")
+	user          = flag.String("user", "", "the database user")
+)
+
+const (
+	createTableSql      = "CREATE TABLE TestAnsiNull (bitcol bit, charcol char(1));"
+	dropTableSql        = "IF OBJECT_ID('TestAnsiNull', 'U') IS NOT NULL DROP TABLE TestAnsiNull;"
+	insertQuery1        = "INSERT INTO TestAnsiNull VALUES (0, NULL);"
+	insertQuery2        = "INSERT INTO TestAnsiNull VALUES (1, 'a');"
+	selectNullFilter    = "SELECT bitcol FROM TestAnsiNull WHERE charcol = NULL;"
+	selectNotNullFilter = "SELECT bitcol FROM TestAnsiNull WHERE charcol <> NULL;"
+)
+
+// This example shows the usage of Connector type
+func ExampleConnector() {
+	flag.Parse()
+
+	if *debug {
+		fmt.Printf(" password:%s\n", *password)
+		fmt.Printf(" port:%d\n", *port)
+		fmt.Printf(" server:%s\n", *server)
+		fmt.Printf(" user:%s\n", *user)
+	}
+
+	connString := fmt.Sprintf("server=%s;user id=%s;password=%s;port=%d", *server, *user, *password, *port)

```

Comment:
I would strongly recommend we use some type of query string builder. In this case, using the net/url.URL type with the Schema set to "sqlserver" would do the trick.